drivers/metal/rendering_device_driver_metal.mm

```diff
@@ -2096,12 +2101,7 @@ void deserialize(BufReader &p_reader) {
 	msl_options.ios_support_base_vertex_instance = true;
 #endif
 
-	bool disable_argument_buffers = false;
-	if (String v = OS::get_singleton()->get_environment(U"GODOT_DISABLE_ARGUMENT_BUFFERS"); v == U"1") {
-		disable_argument_buffers = true;
-	}
-
-	if (device_properties->features.argument_buffers_tier >= MTLArgumentBuffersTier2 && !disable_argument_buffers) {
+	if (device_properties->features.max_buffers_tier >= MTLArgumentBuffersTier2) {
```

Comment:
I moved this logic to `metal_device_properties.mm`, so that it doesn't check each time a shader is compiled.